pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -875,6 +937,44 @@ func (gce *GCECloud) firewallNeedsUpdate(name, serviceName, region, ipAddress st
 	return true, false, nil
 }
 
+func (gce *GCECloud) ensureHttpHealthCheckFirewall(region string, hosts []*gceInstance, hcName string, hcPort int32, isNodesHealthCheck bool) error {
+	// Prepare the firewall params for creating / checking.
+	desc := "Firewall for health checks from load-balancer."
+	sourceRanges := lbSrcRngsFlag.ipn
+	ports := []v1.ServicePort{{Protocol: "tcp", Port: hcPort}}
+
+	clusterId, err := gce.ClusterId.GetId()
+	if err != nil {
+		return fmt.Errorf("error getting cluster ID: %v", err)
+	}
+	fwNameSuffix := MakeHealthCheckFirewallNameSuffix(clusterId, hcName, isNodesHealthCheck)
+	fwName := makeFirewallName(fwNameSuffix)
```

Comment:
Make sense to follow the same naming pattern. I'm using `k8s-{clusterid}-node-hc` for the http healthcheck resource (for healthchecking nodes), we can use it for the corresponding firewall as well.

Since we didn't create firewall for nodeport healthcheck before, following the proposed schema ( `k8s-{clusterid}-{namespace}-{name}-{shortid}-hc`) also seems right. Though I'm not sure should we migrate the old http healthcheck resource (for healthchecking nodeport) to use the new name or should we leave it as is (for now it is something like` aXXXXXXX`)?